datumaro/plugins/synthetic_data/utils.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import warnings
+from random import Random
+
+import cv2 as cv
+import numpy as np
+
+warnings.filterwarnings("ignore", message=r"(invalid value|overflow) encountered")
+
+
+class IFSFunction:
+    NUM_PARAMS = 6
+
+    def __init__(self, rng, prev_x, prev_y):
+        self.function = []
+        self.xs, self.ys = [prev_x], [prev_y]
+        self.select_function = []
+        self.cum_proba = 0.0
+        self._rng = rng
+
+    def add_param(self, params, proba, weights=None):
+        if weights is not None:
+            params = list(np.array(params) * np.array(weights))
+
+        self.function.append(params)
+        self.cum_proba += proba
+        self.select_function.append(self.cum_proba)
+
+    def calculate(self, iterations):
+        prev_x, prev_y = self.xs[-1], self.ys[-1]
+        next_x, next_y = self.xs[-1], self.ys[-1]
+
+        for _ in range(iterations):
```

Comment:
It seems that it is the most computation-intensive part of the algorithm, which can be managed by varying the iteration count. Probably, it would be useful to make it an execution parameter. Is there a reasonable range of values?

Also, I've noticed that some generated images are sharp, while others are blurry. Do you know about the reasons of this?

Reply:
> Also, I've noticed that some generated images are sharp, while others are blurry. Do you know about the reasons of this?

This augmentation allows you to achieve higher accuracy

Reply:
> This augmentation allows you to achieve higher accuracy

Which augmentation?

Please answer on all questions.

Reply:
> It seems that it is the most computation-intensive part of the algorithm, which can be managed by varying the iteration count. Probably, it would be useful to make it an execution parameter. Is there a reasonable range of values?
We chose iterations=200k according to paper

![image](https://user-images.githubusercontent.com/22346860/161045128-66ad6f83-ec5d-4586-a01d-642112fdd87c.png)

Reply:
Augmentation includes flips, rotation and GaussianBlur. Augmentation improved the accuracy of quantization. Details in ticket: 66545
https://github.com/openvinotoolkit/datumaro/blob/5988b950dbb0a84787f7419daca93bf27c1350d6/datumaro/plugins/synthetic_data/utils.py#L131-L147